Src/PChecker/CheckerCore/Specifications/Monitor.cs

```diff
@@ -196,6 +217,36 @@ protected void RaiseEvent(Event e)
             PendingTransition = new Transition(Transition.Type.Raise, default, e);
         }
 
+        /// <summary>
+        /// Writes PrintLog messages to logger.
+        /// </summary>
+        public void LogLine(string message)
+        {
+            Logger.WriteLine($"<PrintLog> {message}");
+
+            // Log message to JSON output
+            JsonLogger.AddLogType(JsonWriter.LogType.Print);
+            JsonLogger.AddLog(message);
+            JsonLogger.AddToLogs(updateVcMap: false);
+        }
+
+        /// <summary>
+        /// Writes message to logger.
+        /// </summary>
+        public void Log(string message)
+        {
+            Logger.Write($"{message}");
+        }
+        
+        /// <summary>
+        /// Stores payload temperarily, and calls <see cref="Monitor.RaiseGotoStateEvent"/>.
+        /// </summary>
+        public void TryGotoState<T>(object payload = null) where T : State
+        {
+            gotoPayload = payload;
+            RaiseGotoStateEvent<T>();
```

Comment:
Change the `RaiseGotoStateEvent` to take a parameter `payload` and then in that function set the gotoPayload field. Then eliminate the TryGotoState function.